src/lib/server/database.ts

```diff
@@ -119,4 +145,8 @@ client.on("open", () => {
 	assistants.createIndex({ modelId: 1, userCount: -1 }).catch(console.error);
 	reports.createIndex({ assistantId: 1 }).catch(console.error);
 	reports.createIndex({ createdBy: 1, assistantId: 1 }).catch(console.error);
+
+	// Unique index for semaphore and migration results
+	migrationResults.createIndex({ guid: 1 }, { unique: true }).catch(console.error);
+	semaphores.createIndex({ key: 1 }, { unique: true }).catch(console.error);
```

Comment:
Should a TTL index be created on `updatedAt` to release the lock in case of crash?

eg `semaphores.createIndex({ updatedAt: 1 },  { expiresAfterSeconds: ... })`

Reply:
Yeah, will use a ttl index instead. We will loose the ability to detect when a lock times out I think  (could be used for rollbacks) but we don't do it anyway at the moment

Reply:
Simplified things, now using a ttl index!